tests/providers/amazon/aws/sensors/test_dynamodb.py

```diff
@@ -0,0 +1,129 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from unittest import mock
+from unittest.mock import MagicMock
+from pprint import pprint
+
+import boto3
+import pytest
+from moto import mock_dynamodb
+
+# from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.dynamodb import DynamoDBHook
+from airflow.providers.amazon.aws.sensors.dynamodb import DynamoDBValueSensor
+
+AWS_CONN_ID = "aws_default"
+REGION_NAME = "us-east-1"
+TABLE_NAME = "test_airflow"
+TASK_ID = "dynamodb_value_sensor"
+
+# breeze testing tests ./tests/providers/amazon/aws/sensors/test_dynamodb.py --log-cli-level=INFO
+
+
+@pytest.fixture
+def use_moto():
+    @mock_dynamodb
+    def dynamodb_client():
+        dynamodb = boto3.resource("dynamodb", region_name=REGION_NAME)
+
+        dynamodb.create_table(
+            TableName=TABLE_NAME,
+            KeySchema=[
+                {"AttributeName": "PK", "KeyType": "HASH"},
+                {"AttributeName": "SK", "KeyType": "RANGE"},
+            ],
+            AttributeDefinitions=[
+                {"AttributeName": "PK", "AttributeType": "S"},
+                {"AttributeName": "SK", "AttributeType": "S"},
+            ],
+            BillingMode="PAY_PER_REQUEST",
+        )
+        return dynamodb
+
+    return dynamodb_client
+
+
+class TestDynamoDBValueSensor:
+    def setup_method(self):
+        self.mock_context = MagicMock()
+
+    def test_init(self):
+        sensor = DynamoDBValueSensor(
+            task_id=TASK_ID,
+            table_name=TABLE_NAME,
+            partition_key_name="PK",
+            partition_key_value="Test",
+            sort_key_name="SK",
+            sort_key_value="2022-07-12T11:11:25-0400",
+            attribute_name="Foo",
+            attribute_value="Bar",
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME,
+        )
+
+        assert TASK_ID == sensor.task_id
+        assert AWS_CONN_ID == sensor.aws_conn_id
+        assert REGION_NAME == sensor.region_name
+
+    @mock_dynamodb
+    def test_get_conn_returns_a_boto3_connection(self):
+        hook = DynamoDBHook(aws_conn_id=AWS_CONN_ID)
+        assert hook.get_conn() is not None
```

Comment:
Here and below, please use `hook.conn` instead of hook.get_conn`.  They are functionally identical but we are trying to work towards standardizing the code when possible.

Reply:
No problem.